app/src/main/java/io/appium/uiautomator2/handler/Source.java

```diff
@@ -30,24 +31,25 @@ public Source(String mappedUri) {
 
     @Override
     public AppiumResponse safeHandle(IHttpRequest request) {
-        ReflectionUtils.clearAccessibilityCache();
-
-        final Document doc = (Document) XMLHierarchy.getFormattedXMLDoc();
-
-        final TransformerFactory tf = TransformerFactory.newInstance();
-        final StringWriter writer = new StringWriter();
-        Transformer transformer;
-        String xmlString;
         try {
-            transformer = tf.newTransformer();
+            ReflectionUtils.clearAccessibilityCache();
+
+            final Document doc = (Document) XMLHierarchy.getFormattedXMLDoc();
+            final TransformerFactory tf = TransformerFactory.newInstance();
+            final StringWriter writer = new StringWriter();
+            Transformer transformer = tf.newTransformer();
             transformer.transform(new DOMSource(doc), new StreamResult(writer));
-            xmlString = writer.getBuffer().toString();
+            String xmlString = writer.getBuffer().toString();
             return new AppiumResponse(getSessionId(request), WDStatus.SUCCESS, xmlString);
+
         } catch (final TransformerConfigurationException e) {
             Logger.error("Unable to handle the request:" + e);
-            throw new RuntimeException("Something went terribly wrong while converting xml document to string", e);
+            return new AppiumResponse(getSessionId(request), WDStatus.UNKNOWN_ERROR, "Something went terribly wrong while converting xml document to string:" + e);
```

Comment:
Same comment from above, do we want to use ```, e)``` instead of ```+ e```

```
return new AppiumResponse(getSessionId(request), WDStatus.UNKNOWN_ERROR, "Something went terribly wrong while converting xml document to string:", e);
```